src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1891,7 +1968,7 @@ protected Void job(final SQLiteConnection sqliteConnection) throws CatalogExcept
           try {
             SQLiteStatement statement =
                 sqliteConnection
-                    .prepare("UPDATE relations SET num_tuples=? WHERE user_name=? AND program_name=? AND relation_name=?;");
+                    .prepare("UPDATE relations SET num_tuples=? WHERE user_name=? AND program_name=? AND relation_name=? and tombstone=0;");
```

Comment:
How could we handle this? Should there be a query before this one which checks if the RelationKey is already tombstoned? If so, then return a ```CatalogException```?